pgzero/actor.py

```diff
@@ -216,6 +224,51 @@ def image(self):
     def image(self, image):
         self._image_name = image
         self._orig_surf = self._surf = loaders.images.load(image)
+        self._adjust_scale_and_angle(self._scale_x, self._scale_y)
+        self._update_pos()
+
+    @property
+    def scale(self):
+        return self._scale_x, self._scale_y
+
+    @scale.setter
+    def scale(self, scale):
+        self._adjust_scale_and_angle(scale[0], scale[1])
+
+    @property
+    def scale_x(self):
+        return self._scale_x
+
+    @scale_x.setter
+    def scale_x(self, x):
+        self._adjust_scale_and_angle(x, 1.0)
+
+    @property
+    def scale_y(self):
+        return self._scale_y
+
+    @scale_y.setter
+    def scale_y(self, y):
+        self._adjust_scale_and_angle(1.0, y)
+
+    @property
+    def size(self):
+        return self._surf.get_size()
+
+    def _adjust_scale_and_angle(self, x, y):
+        if x == 0 or y == 0:
+            raise InvalidScaleException('Invalid scale values. They should be not equal to 0.')
+
+        self._scale_x = x
+        self._scale_y = y
+
+        new_width = int(self._orig_surf.get_size()[0] * abs(x))
+        new_height = int(self._orig_surf.get_size()[1] * abs(y))
+
+        self._surf = pygame.transform.scale(self._orig_surf, (new_width, new_height))
```

Comment:
We should special-case the no-op case for performance for each of these - a check on whether scale_ == scale_y == 1 is much cheaper than doing the transform, which will copy a surface. Same with rotation and flip.

Reply:
done